### ISSUE TITLE
Bull Rush/forced movements and AoOs

### ISSUE THREAD

**Issue by semiliterate** [Vek17/WrathMods-TabletopTweaks#289]
Currently there are two probably-linked bugs with forced movement, particularly that from the force spells such as Battering Blast and Force Punch.

The first is that opponents pushed are subject to Attacks of Opportunity, which I believe is not how tabletop implements forced movement.

The second and more concerning is that opponents pushed get attacks of opportunity on party members near them, making bull rush spells a gamble in one of their traditional roles, pushing people away from wizards before they can attack.

**Comment by Carden-Gixoth** [comment]
Enemies getting AoO when yeeted shouldn't happen. That is a bug, yes. However the [Attacks of Opportunity Rules](https://www.d20pfsrd.com/gamemastering/Combat/#TOC-Attacks-of-Opportunity) (At least those on the SRD) say nothing about forced movement not provoking. This isn't D&D 5e, my guy. It's PF1e, built on D&D 3.5e. However, there is an exception for PF1e. Most yeeting effects like that are considered a Bull Rush Combat Maneuver. Unless you have the Greater Bull Rush Feat, moving the enemy doesn't provoke an AoO from other people. So it comes down to whether yeeting them with Battering Blast and Force Punch are considered Bull Rush attempts. I will check for ya, and get back to you on that.

TL;DR, enemies getting to smack you as they're yeeted is a bug, you getting to smack them _might_ be ok. I will double check and update.

**Comment by Carden-Gixoth** [comment]
OK. I have your update. 

[Battering Blast](https://www.d20pfsrd.com/magic/all-spells/b/battering-blast/) is a Bull Rush attempt, so you need [Greater Bull Rush](https://www.d20pfsrd.com/feats/combat-feats/greater-bull-rush-combat/) to cause enemies to provoke AoOs for being yeeted.

[Force Punch](https://www.d20pfsrd.com/magic/all-spells/f/force-punch/) is an [Overrun](https://www.d20pfsrd.com/gamemastering/combat/#TOC-Overrun) effect, so you actually provoke an Attack of Opportunity for doing so (on top of the one you provoke for casting a spell in Melee), unless you have the [Improved Overrun Feat](https://www.d20pfsrd.com/feats/combat-feats/improved-overrun-combat). 

So, to summarize, you need Greater Bull Rush to have enemies provoke upon being yeeted with Battering Blast, and you need the Improved Overrun feet to not provoke an AoO upon yeeting someone with Force Punch.

You also need to Cast Defensively, or Quicken Spell to not provoke an AoO when casting in Melee, **_period_** cause in PF1e and 3.5e unless you're Casting Defensively, or the Spell is a Swift Action, you _**always**_ provoke an attack of opportunity for casting a spell in a threatened square.

**Comment by semiliterate** [comment]
What's 5e? 3.5 rules, no forced movement provokes :)

If an opponent is in melee, they get an AoO ~after~ the spell triggers, not during its cast phase, so it's not provoked by casting undefensively/failing that check. The AoO doesn't cause a concentration check.

You also need to reread Force Punch, because we're not casting the Mythic version.
